numpy/core/_add_newdocs.py

```diff
@@ -1573,17 +1573,37 @@
         array_function_like_doc,
     ))
 
-add_newdoc('numpy.core.multiarray', '_from_dlpack',
+add_newdoc('numpy.core.multiarray', 'from_dlpack',
     """
-    _from_dlpack(x, /)
+    from_dlpack(x, /)
 
     Create a NumPy array from an object implementing the ``__dlpack__``
-    protocol.
+    protocol. Generally, the returned NumPy array is a read-only view
+    of the input object. See [1]_ and [2]_ for more details.
 
-    See Also
+    Parameters
+    ----------
+    x : object
+        A Python object that implements the ``__dlpack__`` method.
```

Comment:
The spec mandates both `__dlpack__` and `__dlpack_device__` but NumPy just uses the DLTensor to retrieve the device info. So, practically `__dlpack_device__` isn't required to import arrays in NumPy but theoretically, it is. We might want to change it that way then:

```suggestion
        A Python object that implements the ``__dlpack__`` and
        ``__dlpack_device__`` methods.
```

Reply:
Both are required, otherwise there will be issues when consumed by downstream.

Reply:
Added to the docs, thanks @leofang!